plugins/in_splunk/splunk_conn.c

```diff
@@ -27,26 +27,58 @@
 static void splunk_conn_request_init(struct mk_http_session *session,
                                      struct mk_http_request *request);
 
+static void splunk_conn_session_init(struct mk_http_session *session,
+                                   struct mk_server *server,
+                                   int client_fd);
+
+static int splunk_conn_realloc_and_reparse(struct flb_splunk *ctx,
+                                          struct splunk_conn *conn,
+                                          size_t size)
+{
+    char *tmp;
+
+    flb_plg_debug(ctx->ins, "realloc buffer %i -> %zu bytes",
+                  (int)conn->buf_size, size);
+
+    /* Perform realloc */
+    tmp = flb_realloc(conn->buf_data, size);
+    if (!tmp) {
+        flb_errno();
+        flb_plg_error(ctx->ins, "could not perform realloc for size %zu", size);
+        return -1;
+    }
+
+    /* Update buffer info */
+    conn->buf_data = tmp;
+    conn->buf_size = size;
+    /* Keep NULL termination */
+    conn->buf_data[conn->buf_len] = '\0';
+
+    /* Reset parser state */
+    memset(&conn->session.parser, 0, sizeof(struct mk_http_parser));
```

Comment:
looking at the parser code, this memset() is not necessary:

https://github.com/fluent/fluent-bit/blob/master/lib/monkey/include/monkey/mk_http_parser.h#L356